File: 1103 - Distribute Candies to People.cpp
class Solution {
public:
    vector<int> distributeCandies(int candies, int num_people) {
        vector<int> res(num_people);
        int adder = 1;
        while (candies > 0) {
            candies -= adder;
            if (candies >= 0) {
                res[(adder - 1) % num_people] += adder;
            } else {
                candies += adder;
                res[(adder - 1) % num_people] += candies;
                candies = 0;
            }
            adder++;
        }
        return res;
    }
};

File: 1103 - Distribute Candies to People.py
class Solution:
    def distributeCandies(self, candies: int, num_people: int) -> list[int]:
        res = [0] * num_people
        adder = 1
        while candies > 0:
            candies -= adder
            if candies >= 0:
                res[(adder - 1) % num_people] += adder
            else:
                candies += adder
                res[(adder - 1) % num_people] += candies
                candies = 0
            adder += 1
        return res
